### PR TITLE
feat: [#186263393] license status check tab redesign

### PR DESCRIPTION
**Pull request by AnJuHyppolite**
## Description

Complete redesign of the license status check task tab.
**NOTE:** Updated Story Book to account for redesign.

### Ticket

[#186263393](https://www.pivotaltracker.com/story/show/186263393)

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: shared/src/license.ts [hunk 1]
```diff
@@ -9,7 +9,9 @@ export type LicenseStatusResult = {
   checklistItems: LicenseStatusItem[];
 };
 
-export type CheckoffStatus = "ACTIVE" | "PENDING" | "UNKNOWN";
+export type CheckoffStatus = "ACTIVE" | "PENDING" | "INCOMPLETE" | "SCHEDULED" | "NOT_APPLICABLE" | "UNKNOWN";
```
[the remark this answers is not in the document]
Comment (somabadri) [reply shared/src/license.ts:12]: Is there a reason why are we not using the existing license status type?

File: web/src/components/ChecklistTag.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export const ChecklistTag = (props: Props): ReactElement => {
+  const { iconName, iconColor, statusText } = statusObj.get(props.status) ?? {
+    iconName: "",
+    iconColor: "",
+    statusText: "N/A",
+  };
+
+  return (
+    <div className="flex flex-align-center font-sans-2xs width-full tablet:width-auto line-height-sans-2 bg-base-lighter padding-y-05 padding-x-1 padding-right-2 radius-md">
```
Comment (somabadri) [inline web/src/components/ChecklistTag.tsx:32]: [pebble] thoughts on `padding-left-1 padding-right-3` instead of `padding-x-1 padding-right-2` because it might be a little confusing as the `padding-x-1` also contributes to the `padding-right`?

File: shared/src/license.ts [hunk 3]
```diff
@@ -9,7 +9,9 @@ export type LicenseStatusResult = {
...
+
+export type ChecklistStatusText = "Completed" | "Incomplete" | "Pending" | "Scheduled" | "N/A";
```
Comment (somabadri) [inline shared/src/license.ts:14]: [dust] thoughts on moving this into the `ChecklistTag.tsx` file as it's only used there currently and it isn't necessarily shared?

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 4]
```diff
@@ -1,51 +1,47 @@
...
 import { useUserData } from "@/lib/data-hooks/useUserData";
 import analytics from "@/lib/utils/analytics";
 import { LicenseStatus, LicenseStatusItem } from "@businessnjgovnavigator/shared/";
-import { ReactElement, useEffect, useState } from "react";
+import { ReactElement } from "react";
 
 interface Props {
   items: LicenseStatusItem[];
   status: LicenseStatus;
+  agency: string;
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:16]: pebble - if both these props are going to be passed in, why not pass in the entire task?

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 5]
```diff
@@ -1,51 +1,47 @@
...
-const pendingPermitTheme: PermitTheme = {
-  gradient: "gradient-blue",
-  bgColor: "bg-info-extra-light",
-  textColor: "text-cyan",
-  borderColor: "border-cyan",
-  iconColor: "text-base-lighter",
-  headerIconColor: "text-cyan",
-};
 
 const activePermitTheme: PermitTheme = {
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:29]: pebble - the variable name should be updated for all three of these variables, because they no longer describe the theme, they only describe the permit status component

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 6]
```diff
@@ -127,27 +122,24 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
...
-        <Icon className={`${getIconColor(item)} usa-icon--size-3`}>{getIcon(item.status)}</Icon>
-        <span className={`margin-left-2 padding-y-1 fg1 receipt-item ${border}`}>{item.title}</span>
+      <div key={index} data-testid={`item-${item.status}`}>
+        <div className="flex flex-column fac tfr width-full pt-1 tpt-0">
+          <ChecklistTag status={item.status} />
+          <span className="margin-left-2 text-left width-full margin-top-1 tablet:margin-top-0">
+            {item.title}
+          </span>
+        </div>
+        {index === receiptItems.length - 1 ? <></> : <hr className="desktop:margin-bottom-1" />}
```
Comment (somabadri) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:142]: [pebble] Could we wrap this div in another outer div and just add `className="desktop:margin-bottom-1"` so it's easier to read and doesn't require index math?

File: web/src/components/tasks/LicenseTask.tsx [hunk 7]
```diff
@@ -184,6 +184,8 @@ export const LicenseTask = (props: Props): ReactElement => {
                   status={licenseStatusResult.status}
                   items={licenseStatusResult.checklistItems}
                   onEdit={onEdit}
+                  agency={props.task.agencyId ?? ""}
+                  context={""}
```
Comment (somabadri) [inline web/src/components/tasks/LicenseTask.tsx:188]: [pebble] why is the context just an empty string?

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 8]
```diff
@@ -77,48 +73,47 @@ const LicenseStatusLookup: Record<LicenseStatus, string> = {
...
+  "PENDING_REINSTATEMENT",
+]);
+
+const isPending = (licenseStatus: LicenseStatus): boolean => {
+  return pendingStatuses.has(licenseStatus);
+};
+
 export const LicenseStatusReceipt = (props: Props): ReactElement => {
-  const [theme, setTheme] = useState<PermitTheme>(pendingPermitTheme);
+  let theme = pendingPermitTheme;
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:95]: pebble - I don't think Team is inappropriate variable name. It should be updated to clarify what it's actually doing or what area of of the application it's actually impacting

**Review by FarazA22 — CHANGES_REQUESTED** [review]
I'm happy to review this again once the changes are made. If you have any different opinions, feel free to reply back to my comments. Will love your thoughts on some of your decisions. Thanks, Anju.

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 6 above)
Comment (somabadri) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:142]: @FarazA22 I misread this, yeah that makes sense!

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 9]
```diff
@@ -156,35 +148,63 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
...
+
+            <hr className="tablet:margin-bottom-205" />
+
+            {props.items.map(receiptItem)}
+          </div>
+        </div>
+        <HorizontalLine />
+        <div>
+          <strong>{Config.taskDefaults.issuingAgencyText}: </strong>
+          {getAgencyText(props.agency, props.context)}
```
[the remark this answers is not in the document]
Comment (AnJuHyppolite) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:207]: Discussed: The rendering of the `Issuing Agency` is dynamic.

File: content/src/fieldConfig/config.json [hunk 10]
```diff
@@ -213,13 +213,18 @@
     "reinstatementPendingPermitStatusText": "Reinstatement Pending",
     "permitStatusText": "Permit Status:",
     "submittedStatusText": "Submitted",
+    "applicationChecklistItems": "Application Checklist Items",
```
Comment (seidior) [inline content/src/fieldConfig/config.json:216]: [sand] Should this end with the word "Text"? Looking at its siblings, this seems to be a pattern.

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 11]
```diff
@@ -77,34 +73,35 @@ const LicenseStatusLookup: Record<LicenseStatus, string> = {
...
+  "PENDING_REINSTATEMENT",
+]);
+
+const isPending = (licenseStatus: LicenseStatus): boolean => {
+  return pendingStatuses.has(licenseStatus);
+};
+
 export const LicenseStatusReceipt = (props: Props): ReactElement => {
-  const [theme, setTheme] = useState<PermitTheme>(pendingPermitTheme);
+  const [permitColorScheme, setPermitColorTheme] = useState<PermitTheme>(pendingPermitTheme);
```
Comment (seidior) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:95]: [sand] Should this be `permitColorTheme` rather than `permitColorScheme`, given that you use Theme in the setter function and in the default state?

Comment (seidior) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:95]: [pebble] Does this need to be a state variable? Is there a situation you envision that the variable data needs to be stateful and/or changed upon re-render?

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 12]
```diff
@@ -77,34 +73,35 @@ const LicenseStatusLookup: Record<LicenseStatus, string> = {
...
-      setTheme(activePermitTheme);
+      setPermitColorTheme(activePermitTheme);
     } else if (isPending(props.status)) {
-      setTheme(pendingPermitTheme);
+      setPermitColorTheme(pendingPermitTheme);
     } else {
-      setTheme(grayPermitTheme);
+      setPermitColorTheme(inactivePermitTheme);
     }
   }, [props.status]);
```
Comment (seidior) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:106]: [pebble] Likewise, can this be accomplished outside a `useEffect()`?

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 13]
```diff
@@ -127,27 +124,25 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
...
-
-  const receiptItem = (item: LicenseStatusItem, index: number): ReactElement => {
-    const border = index === 0 ? "" : "border-top-1px border-base-light";
-
+  const receiptItem = (
+    item: LicenseStatusItem,
+    index: number,
+    receiptItems: LicenseStatusItem[]
+  ): ReactElement => {
+    console.log("ITEM", item);
```
Comment (seidior) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:136]: [pebble] `console.log()`

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 14]
```diff
@@ -156,35 +151,63 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
...
+              <Icon className="display-none tablet:display-block tablet:margin-left-1 usa-icon--size-4">
+                {getIcon(props.status)}
+              </Icon>
+
+              <p className="tablet:margin-left-1 padding-05 line-height-sans-3">{getText()}</p>
             </div>
           </div>
-          {props.items.map(receiptItem)}
+          <div className="padding-2">
+            <Heading level={1} styleVariant="h4" className="margin-bottom-05-override tablet:padding-top-1">
```
Comment (seidior) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:198]: [dust] Love to see me a `<Heading />`! Just checking though, the desired behavior is a heading level of 1 and a style of `h4-styling`, correct?

**Review by seidior — APPROVED** [review]
This is a massive pull request and some great work, AnJu! I have mostly minor stylistic comments here but overall I think this is a great contribution to the codebase.

File: content/src/fieldConfig/config.json (hunk 10 above)
Comment (AnJuHyppolite) [reply content/src/fieldConfig/config.json:216]: It's such a lengthy key name which is one reason I left it out (I see `reinstatementPendingPermitStatusText`and probably would have left out `text` there had I created that one). Another thought: Essentially these are all text and not every single key ends with text. I am okay with it either way and will make the change.

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 11 above)
Comment (AnJuHyppolite) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:95]: The change should have been from `[theme, setTheme]` to [permitColorScheme, setPermitColorScheme]` based on feedback that the (initial) variable names should "clarify what it's actually doing or what area of of the application it's actually impacting".

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 14 above)
Comment (AnJuHyppolite) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:198]: It's an `<h4>` in the designs, so I added it here as such, but got feedback about it being an accessibility issue since an `<h1>`, `<h2>`, and `<h3>` do not render ahead of the `<h4>`. So I updated it to `<Heading level={1} styleVariant="h4"...` so as not to cause any accessibility issues since there was initially an absence of the higher level tags.

File: web/cypress/e2e/check-license-status.spec.ts [hunk 15]
```diff
@@ -56,7 +56,7 @@ describe("check license status [feature] [all] [group1]", () => {
     cy.get('[data-testid="permit-PENDING"]').should("exist");
     cy.contains("Certificate of CGL Insurance").should("exist");
     cy.contains("Business Formation Documents").should("exist");
-    cy.contains("Aculyst").should("exist");
+    cy.contains("Aculyst".toUpperCase()).should("exist");
```
Comment (FarazA22) [inline web/cypress/e2e/check-license-status.spec.ts:59]: Curious as to why this change was necessary

File: web/src/components/ChecklistTag.tsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    iconColor: "text-base",
+    statusText: Config.licenseSearchTask.incompleteStatusText,
+    statusTextColor: "text-base-darkest",
+    backgroundColor: "bg-base-lightest",
+  },
+  NOT_APPLICABLE: {
+    iconName: "do-not-disturb",
+    iconColor: "text-base",
+    statusText: Config.licenseSearchTask.notApplicableStatusText,
+    statusTextColor: "text-base-darkest",
```
Comment (FarazA22) [inline web/src/components/ChecklistTag.tsx:53]: Since this class is the same across the board, you can remove it here and place it directly in line 68

File: web/src/components/tasks/LicenseTask.tsx [hunk 17]
```diff
@@ -184,6 +184,7 @@ export const LicenseTask = (props: Props): ReactElement => {
                   status={licenseStatusResult.status}
                   items={licenseStatusResult.checklistItems}
                   onEdit={onEdit}
+                  agency={props.task.agencyId ?? ""}
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:187]: Curious why agency is required and potentially passed in with an `"""` but context is optional. I think it should be one or the other not both

**Review by FarazA22 — CHANGES_REQUESTED** [review]

File: web/src/components/tasks/LicenseTask.test.tsx [hunk 18]
```diff
@@ -215,11 +218,11 @@ describe("<LicenseTask />", () => {
 
     it("does not display error alert when license status is found", async () => {
       renderTask();
-
       mockApi.checkLicenseStatus.mockResolvedValue(generateUserData({}));
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:221]: why is this test changing from Pending to Completed? Is seems odd. Either something is wrong or the staged data needs to be explicitly stated in this test.

Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:221]: same comment as above

Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:221]: For this test, nothing has changed but the expected results changed. I initial thought is that it shouldn't change from pending to completed

File: web/src/components/tasks/LicenseTask.test.tsx [hunk 19]
```diff
@@ -215,11 +218,11 @@ describe("<LicenseTask />", () => {
...
       fireEvent.submit(screen.getByTestId("check-status-submit"));
+
       await waitFor(() => {
-        expect(screen.getByText("Pending")).toBeInTheDocument();
+        expect(screen.getByText(Config.licenseSearchTask.completedStatusText)).toBeInTheDocument();
       });
       expect(screen.queryByTestId("error-alert-NOT_FOUND")).not.toBeInTheDocument();
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:227]: For this test, nothing has changed, I don't understand why this is changing from pending to completed text: I'd expect the permit status to continue to display pending. If this is the text within the Tag, I think there should be an assertion around the permit status text and probably rename `completedStatusText` so it's more clear that it's text related to the Tag component

File: web/src/components/tasks/LicenseTask.test.tsx [hunk 20]
```diff
@@ -215,11 +218,11 @@ describe("<LicenseTask />", () => {
 
     it("does not display error alert when license status is found", async () => {
       renderTask();
-
       mockApi.checkLicenseStatus.mockResolvedValue(generateUserData({}));
       fireEvent.submit(screen.getByTestId("check-status-submit"));
+
       await waitFor(() => {
-        expect(screen.getByText("Pending")).toBeInTheDocument();
+        expect(screen.getByText(Config.licenseSearchTask.completedStatusText)).toBeInTheDocument();
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:225]: There are a few places where `pending` is changind to `completed`. They should all be looked at

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 11 above)
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:95]: I agree with Spencer, the variables should be `permitColorTheme` and `setPermitColorTheme`

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 12 above)
Comment (FarazA22) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:106]: I don't think a useEffect is necessary

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 21]
```diff
@@ -114,11 +111,11 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
 
   const getIcon = (status: LicenseStatus): string => {
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:112]: What is this section related to? Is it the Permit section, if so, it should be grouped together

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 13 above)
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:136]: remove console log

File: web/src/components/tasks/LicenseStatusReceipt.tsx [hunk 22]
```diff
@@ -127,27 +124,25 @@ export const LicenseStatusReceipt = (props: Props): ReactElement => {
...
+    index: number,
+    receiptItems: LicenseStatusItem[]
+  ): ReactElement => {
+    console.log("ITEM", item);
     return (
-      <div className="padding-x-3 fdr fac width-100" data-testid={`item-${item.status}`} key={index}>
-        <Icon className={`${getIconColor(item)} usa-icon--size-3`}>{getIcon(item.status)}</Icon>
-        <span className={`margin-left-2 padding-y-1 fg1 receipt-item ${border}`}>{item.title}</span>
+      <div key={index} data-testid={`item-${item.status}`}>
+        <div className="flex flex-column fac tablet-flex-row width-full pt-1 tpt-0">
```
Comment (FarazA22) [inline web/src/components/tasks/LicenseStatusReceipt.tsx:139]: We already have an H1 on this page, setting this to a L1 with create an accessability issue. I personally don't think it should have a heading level unless other areas of the page, like `permit status` have a heading level, you should check with design

File: web/src/components/tasks/LicenseStatusReceipt.tsx (hunk 11 above)
Comment (FarazA22) [reply web/src/components/tasks/LicenseStatusReceipt.tsx:95]: Or, [permitColorScheme, setPermitColorScheme] as you stated in the reply.

File: web/src/components/ChecklistTag.tsx (hunk 16 above)
Comment (AnJuHyppolite) [reply web/src/components/ChecklistTag.tsx:53]: Good catch. Thanks!
